src/engine/contactwriter.cpp

```diff
@@ -2530,6 +2546,64 @@ static ContactWriter::DetailList allSingularDetails()
     return details;
 }
 
+static QContactDetail supportedDetailInstance(QContactDetail::DetailType type)
+{
+    switch (type) {
```

Comment:
'::allSupportedDetails()' in contactwriter.cpp contains the full set of supported detail types.